### ISSUE TITLE
A32NX: Changing Weight on MCDU has no impact on PFD speeds.

### ISSUE THREAD

**Issue by BravoMike99** [flybywiresim/aircraft#9681]
### Aircraft Version

Development

### Build info

```json
{
    "built": "2024-12-18T23:36:26+00:00",
    "ref": "refs/heads/master",
    "sha": "22b58541a8e2e37364a5a1fb757e79fdfa3b1871",
    "actor": "flogross89",
    "event_name": "manual",
    "pretty_release_name": "master:22b58541",
    "version": "a32nx-v0.13.0-dev.22b5854"
}
```

### Describe the bug

Changing the ZFW on the MCDU has no impact on the characteristic speeds on the PFD.

### Expected behavior

FAC characteristic speeds should be calculated based on FM GW (when available) rather than its own internal gross weight model.

### Steps to reproduce

1) Change the ZFW on the MCDU in the air
2) Observe characteristic speeds on PFD.

### Discord Username (optional)

bruno_pt99

**Comment by donstim** [comment]
As an additional comment on this issue (which I've confirmed), the characteristics speeds shown on the MCDU do change when ZFW is changed. It's only the PFD displayed speeds that don't change.

But this doesn't seem like a critical bug to me in that the ZFW should not change once the airplane has left the gate. Both the PFD and MCDU speeds do change as fuel is burned off. The ZFW entered at the start of the flight is used along with the fuel onboard to determine the characteristics speeds, showing that these speeds are indeed based on the FMS GW rather than on an "internal gross weight model."

**Comment by BravoMike99** [comment]
> As an additional comment on this issue (which I've confirmed), the characteristics speeds shown on the MCDU do change when ZFW is changed. It's only the PFD displayed speeds that don't change.
> 
> But this doesn't seem like a critical bug to me in that the ZFW should not change once the airplane has left the gate. Both the PFD and MCDU speeds do change as fuel is burned off. The ZFW entered at the start of the flight is used along with the fuel onboard to determine the characteristics speeds, showing that these speeds are indeed based on the FMS GW rather than on an "internal gross weight model."

I agree that's its a rather niche case but changing the ZFW was the only way I could find to manipulate the FMS GW to be different than the actual weight of the airplane once in the air as we don't model the fuel sensed logic on the fms. For the SLs and NEO the FAC should use the FM GW for the characteristic speeds which means that an erroneous ZFW entry or sensed FOB would reflect itself on the PFD. This is why there's the check weight scratchpad message when the FM GW and FAC calculated GW differ by more than 7T. Currently it must be using the actual weight of the airplane and not the FMS calculated GW as manipulating the ZFW will change the fm gross weight simar which is reflected in other places, such as the the  mcdu speeds as on you mention,  but not on the pfd.

**Comment by donstim** [comment]
> > As an additional comment on this issue (which I've confirmed), the characteristics speeds shown on the MCDU do change when ZFW is changed. It's only the PFD displayed speeds that don't change.
> > But this doesn't seem like a critical bug to me in that the ZFW should not change once the airplane has left the gate. Both the PFD and MCDU speeds do change as fuel is burned off. The ZFW entered at the start of the flight is used along with the fuel onboard to determine the characteristics speeds, showing that these speeds are indeed based on the FMS GW rather than on an "internal gross weight model."
> 
> I agree that's its a rather niche case but changing the ZFW was the only way I could find to manipulate the FMS GW to be different than the actual weight of the airplane once in the air as we don't model the fuel sensed logic on the fms. For the SLs and NEO the FAC should use the FM GW for the characteristic speeds which means that an erroneous ZFW entry or sensed FOB would reflect itself on the PFD. This is why there's the check weight scratchpad message when the FM GW and FAC calculated GW differ by more than 7T. Currently it must be using the actual weight of the airplane and not the FMS calculated GW as manipulating the ZFW will change the FM gross weight including the simvar which is reflected in other places, such as the mcdu speeds as you mention, but not on the FAC speeds/PFD.

Yes, I understand that. We do have the check weight scratchpad message, but you are correct that for some reason the PFD speeds are not using the FMC GW except in one case -- VLS and VAPP when flaps full in approach.
